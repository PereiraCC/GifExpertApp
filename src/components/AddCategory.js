import React, { useState } from 'react'

export const AddCategory = () => {

    const [inputValue, setinputValue] = useState('Hello world');

    const handleInputChange = ( e ) => {
        setinputValue( e.target.value );
    }

    const handleSubmit = ( e ) => {
        e.preventDefault();
    
        console.log('Submit done');
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange }
                />
        </form>
    )
}
