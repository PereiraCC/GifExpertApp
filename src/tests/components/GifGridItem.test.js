import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";


describe('Testing in <GifGridItem />', () => {

    const title = 'A title';
    const url   =  'https://localhost/carlos.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url}/> );
    
    test('must be show <GifGridItem /> correct', () => {
    
        expect( wrapper ).toMatchSnapshot();
    
    });

    test('must have a paragraph with a title', () => {
       
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
    });

    test('must have an image equal to url and alt', () => {
       
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });
    
    test('must have class animate__fadeIn', () => {
       
        const div = wrapper.find('div');
        expect( div.hasClass('animate__fadeIn') ).toBe(true);

    });
});

