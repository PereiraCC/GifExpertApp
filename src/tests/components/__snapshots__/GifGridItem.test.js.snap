// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifGridItem /> must be show <GifGridItem /> correct 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/carlos.jpg"
  />
  <p>
     
    A title
     
  </p>
</div>
`;
